cfme/utils/appliance/__init__.py

```diff
@@ -2434,30 +2434,48 @@ def from_provider(cls, provider_key, vm_name, name=None, **kwargs):
             vm_name: Name of the VM this appliance is running as
             browser_steal: Setting of the browser_steal attribute.
         """
-        from cfme.utils.providers import get_mgmt
-        provider = get_mgmt(provider_key)
+        from cfme.utils.providers import get_crud
+        from cfme.containers.provider.openshift import OpenshiftProvider
+        from cfme.utils import conf
 
-        def is_ip_available():
-            try:
-                ip = provider.get_ip_address(vm_name)
-                if ip is None:
-                    return False
-                else:
-                    return ip
-            except AttributeError:
-                return False
+        provider = get_crud(provider_key)
+        app_kwargs = kwargs.copy()
 
-        if 'hostname' in kwargs:
-            hostname = kwargs.pop('hostname')
-        else:
+        if 'hostname' not in app_kwargs:
+            def is_ip_available():
+                try:
+                    ip = provider.mgmt.get_ip_address(vm_name)
+                    if ip is None:
+                        return False
+                    else:
+                        return ip
+                except AttributeError:
```

Comment:
**Optional**

Consider using `wait_for(handle_exception=True)`, though it isn't limited to only AttributeError exception skipping and may be too broad.